index-map-area.html

```diff
@@ -6,27 +6,62 @@
     <title>index-map-area.html</title>
     <script type="module" src="dist/web-map.js"></script>
     <style>
-      html {
+      html,
+      body {
         height: 100%;
       }
-      body,
-      map {
-        height: inherit;
-      }
       * {
         margin: 0;
         padding: 0;
       }
+      
+      /* Specifying the `:defined` selector is recommended to style the map
+      element, such that styles don't apply when fallback content is in use
+      (e.g. when scripting is disabled or when custom/built-in elements isn't
+      supported in the browser). */
+      map[is="web-map"]:defined {
+        /* Responsive map. */
+        /* max-width: 100%; */
+        
+        /* Full viewport. */
+        /* width: 100%; */
+        /* height: 100%; */
+        
+        /* Remove default (native-like) border. */
+        /* border: none; */
+      }
+      
+      /* Pre-style to avoid FOUC of inline layer- and fallback content. */
+      map[is="web-map"]:not(:defined) + img[usemap],
```

Comment:
I'm a little worried about the complexity of these selectors for users of the polyfill.  I don't recall the gory details at the moment as to why, but I had defined display: none in the connectedCallback of layer.js  for the `<layer->`  with the comment that it avoided displaying inline mapml content but didn't prevent the FOUC; at the moment  I can't really understand what the distinction I was making is.

Reply:
This CSS is totally optional, it is to "automatically" prevent FOUC (with caveat that `<img usemap="">` must come after `<map>` in the DOM order). Defining this in any of the script modules will be too late to prevent FOUC. Of course developers can instead set this themselves, but that leaves it up to each and their own to 1) realize FOUC and 2) come up with styles that prevent FOUC but does not break fallback behavior (which is what we're doing here, _not_ blindly setting client-side image maps to `display: none`).

> I had defined display: none in the connectedCallback of layer.js for the <layer-> with the comment that it avoided displaying inline mapml content but didn't prevent the FOUC

That's still done, only it is now replaced with [this](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/108/files#diff-f073b5f0d60eb22d580b40dd82d5bdf3R98) for `<mm-mapp>` and [this](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/108/files#diff-d6164714af1bf6c21f54c11ca3e1a588R108) for `<map>`, which has the benefit in hiding not only layer elements, but all fallback content (since this is set from the script module, custom elements is supported and we can safely hide any potential fallback child elements).

I suppose, working with custom elements will never be as pretty as native ones 😉

Reply:
> with caveat that <img usemap=""> must come after <map> in the DOM order

That I didn't know either.  Do you mean to prevent FOUC, or just generally AND it prevents FOUC.  

> > I had defined display: none in the connectedCallback of layer.js for the  with the comment that it avoided displaying inline mapml content but didn't prevent the FOUC
> 
> That's still done, only it is now replaced with [this](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/108/files#diff-f073b5f0d60eb22d580b40dd82d5bdf3R98) for `<mm-mapp>` and [this](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/108/files#diff-d6164714af1bf6c21f54c11ca3e1a588R108) for `<map>`, which has the benefit in hiding not only layer elements, but all fallback content (since this is set from the script module, custom elements is supported and we can safely hide any potential fallback child elements).

Very nice solution.  This PR is an eye-opener, thank you.  I'll keep reading.  

By the way, I have been noodling about how to actually implement a better polyfill, one that actually contains DOM nodes that the author can interact with.  I was thinking that we could walk the DOM tree and attach the mapml.js layer subtype objects to the appropriate nodes.  The resulting events and so on might make more sense in such a context.  Not sure if I'm describing this right, but imagine you have a  `<layer- ...><extent><input type="location" ...>...<link tref="..." ...></extent></layer->`. Instead of creating an `<extent->` or equivalent `<mapml-extent>` custom element, you just parse the DOM nodes that you're given and attach the appropriate mapml (sub) layer type to the appropriate node.  The mapml layer type continues to render itself in the shadow DOM, but maybe you have a better chance of simulating the behaviour of the node-based existence of the render item.  

We can discuss this outside of this PR, I just wanted to ask you your opinion.  Thanks

Reply:
> > with caveat that  must come after in the DOM order
> 
> That I didn't know either. Do you mean to prevent FOUC, or just generally AND it prevents FOUC.

_Only_ to prevent FOUC, but we still want to ensure it's hidden if the browser supports custom elements, which you're already doing in (I haven't removed): https://github.com/Maps4HTML/Web-Map-Custom-Element/blob/592dc2ae9e9536b676aa4b399c89357a1fe70ec0/src/web-map.js#L174-L178